exec/java-exec/src/main/java/org/apache/drill/exec/dotdrill/View.java

```diff
@@ -127,6 +135,8 @@ public FieldType(String name, RelDataType dataType) {
       default:
         break;
       }
+      keyType = dataType.getKeyType() == null ? null : new FieldType(null, dataType.getKeyType());
+      valueType = dataType.getValueType() == null ? null : new FieldType(null, dataType.getValueType());
```

Comment:
I confused Field Type Name with Filed Name. Filed Name should be used for `FiledType`. The name of `FiledType` is confusing. Consider renaming `FiledType` to `Filed`.
Regarding null in constructor, I agree there is no need to store the key, value names for every internal MAP (it will be not a view, but dataset then).
But I think it will be good to introduce the new constructor without Field Name. 
This constructor will serve for representing internal Fields of complex Fields (MAP now, maybe List in future) in the View.